File: 0977-squares-of-a-sorted-array/0977-squares-of-a-sorted-array.cpp
class Solution
{
public:
    vector<int> sortedSquares(vector<int> &nums)
    {
        vector<int> res;

        int left = 0, right = nums.size() - 1;
        while (left <= right)
        {
            int numR = abs(nums[right]);
            int numL = abs(nums[left]);
            if (numR > numL)
            {
                res.push_back(numR * numR);
                right--;
            }
            else
            {
                res.push_back(numL * numL);
                left++;
            }
        }
        reverse(res.begin(), res.end());
        return res;
    }
};

File: 0977-squares-of-a-sorted-array/0977-squares-of-a-sorted-array.java
class Solution {
    public int[] sortedSquares(int[] nums) {
        int[] res = new int[nums.length];
        int left = 0;
        int right = nums.length - 1;
        int i = nums.length-1;
        while (left <= right) {
            int rightNum = Math.abs(nums[right]);
            int leftNum = Math.abs(nums[left]);

            res[i--] = (int) Math.pow(Math.max(leftNum, rightNum),2);

            if (leftNum <= rightNum)
                right--;
            else
                left++;
        }
        return res;
    }
}